backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -297,12 +296,12 @@ private class ExportedElement(val kind: ElementKind,
             original is ConstructorDescriptor ->
                 SignatureElement(uniqueName(original, shortName), original.constructedClass.defaultType)
             else ->
-                SignatureElement(uniqueName(original, shortName), original.returnType!!)
+                SignatureElement(uniqueName(original, shortName), original.returnType!!.noInlineUnderlyingRepresentation() ?: original.returnType!!)
```

Comment:
What is the purpose of this patch? Does it change how types are mapped?

Reply:
I see.
New mapping behaviour seems incorrect. For `takeNullableBar` from [the motivational test case](https://github.com/JetBrains/kotlin-native/pull/4200/commits/ac0a57c3f7df6cc3d7834bad2ade56ae5807d701#r434619545) it should produce `Bar` as before, not `Any`.

Mapping was ok, it is generation of used types that was broken.

Reply:
Hm, I'm totally confused now. current (master) behaviour is: avoid type generation for inline classes. and my patch prevent appearing "unwanted" types in return and as parameter. Or I incorrectly understand the idea?